pageserver/src/walrecord.rs

```diff
@@ -87,6 +127,28 @@ impl XlRelmapUpdate {
     }
 }
 
+#[repr(C)]
+#[derive(Debug)]
+pub struct XlSmgrCreate {
+    pub rnode: RelFileNode,
+    // FIXME: This is ForkNumber in storage_xlog.h. That's an enum. Does it have
+    // well-defined size?
```

Comment:
32 bit

Reply:
Yes, size of enum is compiler specific (as almost everything in C:),
but practically I do not know C compilers where sizeof(enum) != sizeof(int) if all enum constants fits in int range. Yes, you can use compiler options and pragmas to change enum size. But I will not be surprised that attempt to rebuild postgres with -fshort-enums will cause not only incompatibility with WAL/data created by normal version of postgres, but cause crashes or incorrect work of such postgres.

There is actually only one place in postgres where ForkNumber is used in persistent structure (which is expected to be saved on disk) -  xl_smgr_create. And even here size of structure will not changes even if sizeof(ForkNumber) != sizeof(int) because of alignment.
Actually Postgres doesn't pretend that it WAL/data file format is machine independent. So I do not see some new/real problem here.